### DISCUSSION TITLE
How to Client Side redirect without page reload

### DISCUSSION THREAD

**Discussion by remibeaufils** [vercel/next.js#20625] [Help]
How to make a Client Side redirect without a page reload?

I'm using next v10.0.4.
Inside a component function I know I can do that (see code below).
But I need to perform a redirect outside a component function so I can't use the useRouter hook.

```js
const AboutPage: NextPage<Props> = ({ owners }) => {
  // With this technique, the redirect happens and the page doesn't reload. ✅
  const router = useRouter();
  router.replace(`login?page_requested=user/1`);

  // With this technique, the redirect happens but the page reload. ❌
  Router.prototype.replace(`login?page_requested=user/1`);
  
  return (
    ...
```

**Comment by lachlanjc** [comment]
Check out this comment! https://github.com/vercel/next.js/issues/11797#issuecomment-621538365
